sure/__init__.py

```diff
@@ -925,6 +925,13 @@ def match(self, regex, *args):
 expect = AssertionBuilder('expect')
 
 
+def assertion(func):
+    """Extend sure with a custom assertion method."""
+    func = assertionmethod(func)
+    setattr(AssertionBuilder, func.__name__, func)
```

Comment:
Maybe don't use `setattr` but registration of assertion into a dict and then use `__getattr__` ...?